src/main/java/com/puppycrawl/tools/checkstyle/meta/MetadataGeneratorUtil.java

```diff
@@ -74,8 +107,190 @@ public static void generate(String path, OutputStream out, String... moduleFolde
         checker.addListener(new MetadataGeneratorLogger(out, OutputStreamOptions.NONE));
 
         final List<File> moduleFiles = getTargetFiles(path, moduleFolders);
+        final List<File> moduleFilesForNewMetadata = moduleFiles.stream().filter(file -> {
+            final String fileName = file.getName();
+            return MODULES_GENERATED_BY_NEW_FUNCTIONALITY.contains(fileName);
+        }).collect(Collectors.toList());
+        moduleFiles.removeAll(moduleFilesForNewMetadata);
 
         checker.process(moduleFiles);
+
+        //////////////////////////////
+        // Plan:
```

Comment:
please move this block of code to new method.

Reply:
Done.